core/dbt/parser/manifest.py

```diff
@@ -1693,7 +1695,11 @@ def write_semantic_manifest(manifest: Manifest, target_path: str) -> None:
 
 
 def write_manifest(manifest: Manifest, target_path: str):
-    path = os.path.join(target_path, MANIFEST_FILE_NAME)
+    if get_command_name() == "unit-test":
+        file_name = UNIT_TEST_MANIFEST_FILE_NAME
+    else:
+        file_name = MANIFEST_FILE_NAME
+    path = os.path.join(target_path, file_name)
```

Comment:
You mean set a field in the unit_test task?  "write_manifest" is called from the run method in runnable.py.

Reply:
"write_manifest" is called in the outer_wrapper/wrapper method in core/dbt/cli/requires.py. It wouldn't have access to something set in the unit test task...

Reply:
If we make the unit_test_manifest a different class of object, we could check that... Maybe we can defer this until after we look at that change?

Reply:
Perhaps we could use `flags.which` in the requires wrapper. It still is routing based on the command which doesn't feel great but in a less deeply nested part of the code. Looks like we're already doing something similar (accessing flags) in that wrapper as well: https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/cli/requires.py#L252-L253

Reply:
Actually it turns out that we don't want to write out the unit test manifest in the cli requires wrapper function, because that's the full parsed manifest.  We want it written out in the runnable.py code.

Reply:
Done.